widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.java

```diff
@@ -986,10 +1000,6 @@ private void showEndEngagementDialog(String operatorName) {
                     if (controller != null) {
                         controller.endEngagementDialogYesClicked();
                     }
-                    if (onEndListener != null) {
```

Comment:
Survey is optional? What will happen if engagement should end without survey

Reply:
It handles in ChatController.
Anyway, onEndListener.onEnd() calls from the destroyView().